### DISCUSSION TITLE
How to support DateOnly/TimeOnly

### DISCUSSION THREAD

**Discussion by truthz03** [Mimetis/Dotmim.Sync#1263] [Q&A]
Hi. I'm using some DateOnly and TimeOnly properties in my EntityFrameworkCore database context.

My server is using MSSql and there the data is saved as "date".
If i sync it to my frontend with Sqlite it gets a TEXT colum. I think this is normal because EntityFramework migrations also create it as TEXT.

The problem is that in this TEXT colum the value will be "2024-06-07 00:00:00" after the sync.
If I trie to load the data now with EntityFramework a get an error because this string can not be casted to DateOnly.

If I manually delete the time inside the Sqlite db I works.

Is it possible to remove the time inside the sync process?
I already tried with a custom converter but this doesn't work

**Comment by Mimetis** [comment]
`DateOnly` has been added recently in **DMS**, if you are using **.NET 6** or higher.

I think we don't have any tests on the compatibility between **SQL Server** and the synced type in **SQLite**
I will make some tests and see what I can do.

If you have already a really small sample to test it, would be cool, otherwise I'll work on it.

By the way, did you try to use EF converter ?

``` cs
public class YourDbContext : DbContext
{
    Ctors() {...}

    protected override void ConfigureConventions(ModelConfigurationBuilder builder)
    {
        builder.Properties<DateOnly>()
                .HaveConversion<DateOnlyConverter>()
                .HaveColumnType("date");
    }
}
```

**Reply by truthz03** [reply]
Sorry I missed to tell you my used DMS version. I'm using Net8 and DMS 1.0.2 because 1.2.0 is currently not working with my software.
Should `DateOnly` also be supported by 1.0.2?

Sorry my software is really big because my software was already "finished" but because of some problems keeping my db in sync I try to switch to DSM now. Therefore I can't give you a small sample.

No, I have not tried it with an EF converter but I will try now.

**Reply by truthz03** [reply]
I tried 1.2.0 und 1.3.0-beta now again it is working now.
I don't know why 1.2.0 was not working at the beginning of my tests.
The only thing I changed was to add an empty string as schema to all methods, maybe this was the problem before.

With 1.2.0 and 1.3.0 the DateOnly was converted correctly without adding any additional converter :)

**Reply by truthz03** [reply]
rejoiced too soon

With DMS version 1.2.0 I get this error:
sanity check on input stream failed, json token gap of more than 1048576 bytes

With DMS version 1.3.0-beta it seems to work again but I don't know if it is a good idea to use this in production ^^

Any idea what I can do or did you know when 1.3.0 will be released?

**Reply by Mimetis** [reply]
Yes, you can, `1.3.0` is just the latest version and I will release it as official version very soon.
`1.3.0` is correcting some issues from `1.2.0`
